src/Cart.java

```diff
@@ -10,9 +10,27 @@ public Cart() {
         this.productsInCart = new ArrayList<>();
     }
 
+    //metodo geralizzato che verifica prima se si può finalizzare la vendita
+    // e poi procede svuotando il carrello e rimuovendo i device venduti dal magazzino;
+
+    public static void finalizeSellingProcess(boolean paymentCheck, List<Product> productsInCart, List<Product> productsInStock) throws IllegalAccessException {
```

Comment:
productsInCart essendo già nella classe non cè bisogno di passarlo come parametro. Anche productsInStock non dovrebbe essere necessario passarlo come parametro, perchè dovresti essere in grado di richiamare direttamente Warehouse. L'unico parametro necessario dovrebbe essere appunto il boolean.

Reply:
è solo il nome ma è un metodo, gli devo per forza dare in ingresso le liste con cui lavorare e non ha senso crearmi il carrello ed il magazzino nel metodo stesso, non capisco cosa vuoi che faccia .

Reply:
Parto col presupposto che questa cosa l' ho fatta in contesto diverso, ma dovrebbe funzionare ugualmente anche qua. Se mai ci vediamo un attimo in separata sede se dovessi avere problemi. Dovresti fare la stessa cosa che fai, solo che invece che fare le operazioni che fai con i parametri che inserisci nel metodo, le fai direttamente con il vero e proprio carrello ed il vero e proprio warehouse. Altrimenti ogni volta che richiami il metodo ci devi per forza passare due liste come parametro, quando queste liste dovrebbero già essere definite a priori, non variabili in base a che lista ci passi al metodo. Però ripeto non ho la certezza 100% che si possa fare finchè non lo scrivo e non incorro in nessun problema.